### PR TITLE
Add encode/decode options to create store functions

### PR DESCRIPTION
**Pull request by konclave**
Motivation: only primitive values can be stored in localStorage. If I want to persist some complex value like an array of objects, I need a way to convert it to string and back.
encode and decode optional functions processes the value before storing it in the storageEngine and after reading the value from storageEngine.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { createMap, createStore } from 'nanostores'
 
+let id = a => a
```
Comment (ai) [inline index.js:3]: Let’s make more reable name

**Comment by ai** [comment]
I like this change. Let’s fix the name and we can merge it.

File: index.js (hunk 1 above)
Comment (konclave) [reply index.js:3]: That's the [identity function](http://www.zvon.org/other/haskell/Outputprelude/id_f.html) name from the Haskell standard Prelude library =)
I'll rename it to `identity`.

**Comment by ai** [comment]
Thanks. Released in 0.3.

Do you have Twitter account to mention you in release tweet?

**Comment by konclave** [comment]
Yes. https://twitter.com/konclave

**Pull request merged**